File: Linguistics/Orthographies/CasedOrthography.vb
''' <summary>
''' Represents an <see cref="Orthography"/> with casing.
''' </summary>
Friend NotInheritable Class CasedOrthography
	Inherits Orthography

	Private ReadOnly UppercaseGlyphs As IReadOnlyCollection(Of String)

	Private ReadOnly LowercaseGlyphs As IReadOnlyCollection(Of String)

	Friend Sub New(UppercaseGlyphs As String(), LowercaseGlyphs As String())
		Me.UppercaseGlyphs = UppercaseGlyphs
		Me.LowercaseGlyphs = LowercaseGlyphs
	End Sub

	Public Overrides ReadOnly Property Count As Int32
		Get
			Return UppercaseGlyphs.Count + LowercaseGlyphs.Count
		End Get
	End Property

	Public Overrides Function GetEnumerator() As IEnumerator(Of String)
		Return UppercaseGlyphs.Concat(LowercaseGlyphs).GetEnumerator()
	End Function
End Class

File: Literary/Orthographies/CasedOrthography.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Stringier.Literary {
	/// <summary>
	/// Represents an <see cref="Orthography"/> with casing
	/// </summary>
	internal sealed class CasedOrthography : Orthography {
		private IReadOnlyCollection<Glyph> UppercaseGlyphs;

		private IReadOnlyCollection<Glyph> LowercaseGlyphs;

		internal CasedOrthography(String[] uppercaseGlyphs, String[] lowercaseGlyphs) {
			UppercaseGlyphs = uppercaseGlyphs.ToGlyphs();
			LowercaseGlyphs = lowercaseGlyphs.ToGlyphs();
		}

		/// <inheritdoc/>
		public override Int32 Count => UppercaseGlyphs.Count + LowercaseGlyphs.Count;

		/// <inheritdoc/>
		public override IEnumerator<Glyph> GetEnumerator() => UppercaseGlyphs.Concat(LowercaseGlyphs).GetEnumerator();
	}
}
